types/sanctuary/index.d.ts

```diff
@@ -48,28 +49,31 @@ type NonNegativeInteger     = number;
 
 interface TypeRep {}
 
-interface Setoid<A> {}
-interface Ord<A> extends Setoid<A> {}
-interface Semigroupoid<A, B> {}
-interface Category<A> extends Semigroupoid<A, A> {}
-interface Semigroup<A> {}
-interface Monoid<A> extends Semigroup<A> {}
-interface Functor<A> {}
-interface Bifunctor<A, C> extends Functor<C> {}
-interface Profunctor<B, C> extends Functor<C> {}
-interface Apply<A> extends Functor<A> {}
+interface Alt<A> extends Functor<A> {}
+interface Alternative<A> extends Applicative<A>, Plus<A> {}
 interface Applicative<A> extends Apply<A> {}
+interface Apply<A> extends Functor<A> {}
+interface Bifunctor<A, C> extends Functor<C> {}
+interface Category<A> extends Semigroupoid<A, A> {}
 interface Chain<A> extends Apply<A> {}
 interface ChainRec<A> extends Chain<A> {}
+interface Comonad<A> extends Extend<A> {}
+interface Contravariant<A> {}
+interface Extend<A> extends Functor<A> {}
+interface Foldable<A> {}
+interface Functor<A> {}
+interface Group<A> extends Monoid<A> {
```

Comment:
@davidchambers `Group<A>` is the only new Type Class

Reply:
Used for [`S.invert`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59923/files#diff-5f5b52c902d2832b88260d5f71d8957cd9fd861a6cc3b4a65b5dc92c4af08b9fR120)

Reply:
Could we preserve the current ordering, and simply insert `Group<A>` after `Monoid<A>`?

Reply:
Sure but I'm curious to why you do not prefer alphabetic ordering? Ahh I see, you are using the ordering from https://github.com/sanctuary-js/sanctuary-type-classes#type-class-hierarchy, right?
I'll add a commit that we retain the original order.